packages/components/src/box-control/utils.ts

```diff
@@ -272,3 +273,62 @@ export function getAllowedSides(
 	} );
 	return allowedSides;
 }
+
+/**
+ * Checks if a value is a preset value.
+ *
+ * @param value     The value to check.
+ * @param presetKey The preset key to check against.
+ * @return Whether the value is a preset value.
+ */
+export function isValuePreset( value: string, presetKey: string ) {
+	return value.startsWith( `var:preset|${ presetKey }|` );
+}
+
+/**
+ * Returns the index of the preset value in the presets array.
+ *
+ * @param value     The value to check.
+ * @param presetKey The preset key to check against.
+ * @param presets   The array of presets to search.
+ * @return The index of the preset value in the presets array.
+ */
+export function getPresetIndexFromValue(
+	value: string,
+	presetKey: string,
```

Comment:
As far as I know, it is the first time we have introduced support for dealing with preset variables inside the component packages. In other cases e.g.: colors, font,s etc, the component just deals with values and something outside the component packages takes care of converting between values and variable presets.


Is there any reason why we opted in this case to bring the variable conversion inside the components package?

Reply:
Yes, there's a reason. I don't think setting a "preset" is the same thing as setting a "value" that is equivalent to the preset. I explain this in the PR description

> If I switch back and forth between presets and custom value, I don't try to "guess" the value from the preset or the opposite, this is a conscious decision because for me, setting a preset who's current value is 4px is different than setting 4px custom value. The preset value can change over time.

I think the other alternative to use the special value annotation is using some kind of `{ type: 'preset', slug }` as a value. I hesitated but I felt that since we're already using the anotation heavily everywhere and it has proven to be a good solution, we should just be consistent and use it instead of inventing yet another "shape" of value.

Reply:
Got it `{ type: 'preset', slug }` seems cleaner and more generic for components packages, e.g: I may prefer to apply presets as classes or other solutions, but you are right our variable notation is widely spread and using this cleaner notation would just involve having wrappers doing the conversion for the variable notation somewhere and more code for us to maintain.

Reply:
I wish these preset layers could be implemented downstream in consumerland, on top of wishing that BoxControl lived in wp-block-editor to begin with, but given the reality I agree it's inevitable that we implement it in wp-components.

Reply:
I share Lena's sentiment. Adding presets to me feels like something that should not belong to the components package, but I understand the compromise — and I also, like Lena, think that `BoxControl` should belong to the block editor, but that ship has already sailed.

Also, if we think that we'll want to use presets more widely, it may make sense to extract some of their logic to a separate hook / utility functions.